From 8ba4fb1c21222dd1e6c0cfdb91cb6e201cdd35d0 Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Fri, 02 Oct 2020 23:03:22 +0300
Subject: [PATCH] [SW-21029] ppboot: goya: add support for bss

This patch adds support for bss section in ppboot
such that SW can use global variables. Further, they
will be cleared in crt file before jumping to
ppboot_main. These were required to be added in
Goya2 and hence the same is being ported to
goya/gaudi as well. It’s needed to prevent compilation issues

Change-Id: I34bfbc5d42916a57fca8ad01e7fd89f580204d3f
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/goya/ppboot.lds.S b/goya/ppboot.lds.S
index e078b30..1e5be88 100644
--- a/goya/ppboot.lds.S
+++ b/goya/ppboot.lds.S
@@ -20,6 +20,13 @@
 		* (.text*)
 	} >.sram
 
+	.bss : {
+		. = ALIGN(8);
+		__bss_start = . ;
+		*(.bss*)
+		__bss_end = . ;
+	} > .sram
+
 	.stack : {
 		. = ALIGN(8);
 		__stack = . ;
